### PR TITLE
Fix `Kosmos2Processor` batch mode

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

Fix `Kosmos2Processor` batch mode as there is a bug, see comment in the code change.

Fix one issue about batch mode opened in #27301

### CODE REVIEW

File: src/transformers/models/kosmos2/processing_kosmos2.py [hunk 1]
```diff
@@ -211,7 +211,9 @@ def __call__(
                 image_embeds_position_mask.append(mask)
 
             if isinstance(text, list):
-                sorted_length = sorted([(idx, len(x)) for idx, x in enumerate(text_encoding.input_ids)])
+                sorted_length = sorted(
+                    [(idx, len(x)) for idx, x in enumerate(text_encoding.input_ids)], key=lambda x: x[-1]
```
Comment (ydshieh) [inline src/transformers/models/kosmos2/processing_kosmos2.py:215]: Here my intention was to sort with the length, not the index. However, I forgot to specify the `key` so it did not do what I expected.

Comment (ydshieh) [reply src/transformers/models/kosmos2/processing_kosmos2.py:215]: Note that we do have tests regarding the batch mode - but the examples I used have the longest sequence as the last example in a batch, so it didn't detect this issue 😭

**Comment by ydshieh** [comment]
All tests (even slow) pass!

File: src/transformers/models/kosmos2/processing_kosmos2.py (hunk 1 above)
Comment (amyeroberts) [reply src/transformers/models/kosmos2/processing_kosmos2.py:215]: Can we update the test so that it would have been caught?

**Review by amyeroberts — APPROVED** [review]
Thanks for fixing this!

File: src/transformers/models/kosmos2/processing_kosmos2.py (hunk 1 above)
Comment (ydshieh) [reply src/transformers/models/kosmos2/processing_kosmos2.py:215]: OK!

**Comment by rabiulcste** [comment]
@ydshieh thanks for the fix! any thoughts on the multi-GPU issue?

**Pull request merged**